### PR TITLE
add package revoking logic

### PR DESCRIPTION
**Pull request by jjhelmus**

### CODE REVIEW

File: conda_build/index.py [hunk 1]
```diff
@@ -289,6 +289,9 @@ def _read_index_tar(tar_path, lock, locking=True, timeout=90):
         with tarfile.open(tar_path) as t:
             try:
                 index_json = json.loads(t.extractfile('info/index.json').read().decode('utf-8'))
+                # add unsatisfiable dependency to revoked packages
+                if isfile(tar_path + '.REVOKED'):
+                    index_json['depends'].append('package_has_been_revoked')
```
Comment (kalefranz) [inline conda_build/index.py]: I can cave on this solution for backward compatibility, but I don't want it as a requirement moving forward.  Can we also add:

    index_json['revoked'] = True

?

**Comment by kalefranz** [comment]
After looking at this again, I'm not sure it'll do what you expect...

Lines 215-219 of index.py have early-exit if the the metadata from the package has already been read up once.  The best place to do the revoked file check is probably going to be around those lines of code.

**Comment by jjhelmus** [comment]
@kalefranz Thanks for the pointer to the early exit.  5256dab adds logic so that packages are always indexed when the REVOKED file and the index disagree.  Adding (or removing) and REVOKED file will trigger the file to be re-indexed.

**Comment by kalefranz** [comment]
Haven't tested this myself, but it looks like it should work.  Guess when we run it the first time, we should double check and make sure it's working as intended.

**Pull request merged**